tests/host_tools/readmem.c

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+
+#define MB (1024 * 1024)
+
+
+int read_mem(int mb_count, int value) {
+    int i;
+    char *ptr = NULL;
+    int *cur = NULL;
+
+    do {
+        ptr = mmap(
+            NULL,
+            mb_count * MB * sizeof(char),
+            PROT_READ | PROT_WRITE,
+            MAP_ANONYMOUS | MAP_PRIVATE,
+            -1,
+            0
+        );
+    } while (ptr == MAP_FAILED);
+        
+    cur = (int *) ptr;
+    for(i = 0; i < (mb_count * MB * sizeof(char)) / (4 * sizeof(int)); i += 4) {
+        if (cur[i] == value &&
```

Comment:
If I understand this correctly, this loop returns `1` when a memory area consisting of 4 consecutive `int`s are equal to `value`.

You could make it quicker by filling a buffer with `value` and using `memcmp` to compare where `cur` points to the buffer.

Reply:
That's true, I changed it to use a buffer and `memcmp`.